### PR TITLE
Support declarative event filters

### LINKED ISSUE #2: Missing event filters
This library doesn't seem to support event filters, which can intercept an event during the event capturing phase (which is when it's being propagated from root to target). See:

https://docs.oracle.com/javafx/2/events/processing.htm

### PR DESCRIPTION
**Pull request by frenchy64**
Adds extension lifecycle `cljfx.ext.node/with-event-filter-props` to declaratively specify event filters.

Fixes #2

### CODE REVIEW

**Comment by vlaaad** [comment]
Hey, that's a cool example!
I initially thought about having declarative event filters as a map from event type to handler or list of handlers, but since there is a catch-all "any" event, and you can check your event type in handler, this is quite an elegant solution! 

I think it should not be an extra-props lifecycle, but instead a part of a core node props in `cljfx.fx.node/props`. 2 main use cases for extra-props lifecycles are:
- adding props that I missed without having to patch cljfx itself when you are writing an application and just want to get stuff done
- adding synthetic props which are not present on mutable objects as a "place", such as selection props: they are part of selection mode, but we pretend it is a part of a list/table because it makes sense.

Event filter feels more like an essential part of a Node in that regard, that's why I think it should be a part of node props.

**Comment by frenchy64** [comment]
Thanks for your review. I moved the functionality to the `:event-filter` prop of `fx.node`.

I initially used an extension lifecycle to make a small library based on the devtools example that "instrumented" your scene automatically by passing it a description of your root node. With this convenient Node prop, perhaps it's better exposing the functionality as an event filter "handler", ie. a library providing the MouseEvent logic in the example.

File: src/cljfx/fx/node.clj [hunk 1]
```diff
@@ -24,6 +27,13 @@
     :disable [:setter lifecycle/scalar :default false]
     :effect [:setter lifecycle/dynamic]
     :event-dispatcher [:setter lifecycle/scalar]
+    ; takes an event description that receives filtered Events with event type Event/ANY
+    :event-filter (prop/make
+                    (mutator/adder-remover
+                      #(.addEventFilter ^Node %1 Event/ANY ^EventHandler %2)
+                      #(.removeEventFilter ^Node %1 Event/ANY ^EventHandler %2))
+                    lifecycle/event-handler
+                    :coerce coerce/event-handler)
```
Comment (vlaaad) [inline src/cljfx/fx/node.clj]: Sorry for nitpicking, but I want a minor correction here: `composite/props` macro transforms vector values to `prop/make` calls to reduce boilerplate, and thus preferred over `prop/make` calls:
```clj
[(mutator/added-remover ...)
 lifecycle/event-handler
 :coerce coerce/event-handler]
```

**Comment by frenchy64** [comment]
Updated to vector style, and updated example use a pure[r] helper function like I suggested above.

**Comment by vlaaad** [comment]
Looks good, I'll try it at home one more time later today and hopefully will merge it!

**Comment by vlaaad** [comment]
Thanks!

**Pull request merged**